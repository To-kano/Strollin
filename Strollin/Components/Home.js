import React from 'react';
import {
  View, Button
} from 'react-native';
import { connect } from 'react-redux';
import I18n from '../Translation/configureTrans';

function Home(props) {
  return (
    <View style={{ flex: 1 }}>
      <View style={[{ flex: 1, backgroundColor: 'white' }]} />
      <View style={{ flex: 1.5, marginHorizontal: '35%', backgroundColor: 'white' }}>
        <Button
          title={I18n.t('Home.historic')}
          color="#89B3D9"
          onPress={() => props.navigation.navigate('historicUser')}
        />
        <Button
          title={I18n.t('Home.logOut')}
          color="#89B3D9"
          onPress={() => props.navigation.navigate('userLogin')}
        />
      </View>
      <View style={[{ flex: 0.5, backgroundColor: 'white', justifyContent: 'center' }]}>
        <Button
          color="#D99860"
          title={I18n.t('Home.newTrip')}
          onPress={() => props.navigation.navigate('TripSuggestion')}
        />
      </View>
    </View>
  );
}

//const mapStateToProps = (state) => state;

//export default connect(mapStateToProps)(Home);
export default Home;
