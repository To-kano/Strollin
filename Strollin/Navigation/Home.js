import React from 'react';


import { createStackNavigator, TransitionPresets } from '@react-navigation/stack';

// import Home from "../Components/HomeScreen";
import Profile from '../Components/ProfileScreen';
// import ConnectionScreen from '../Components/ConnectionScreen';
import UserRegister from '../Components/UserRegister';
import PartnerRegister from '../Components/PartnerRegister';
import TagSelection from '../Components/TagSelection';
import LoginPage from '../Components/LoginPage';
import FriendList from '../Components/FriendList';
import HomePage from '../Components/HomePage';
import TripSuggestion from '../Components/TripSuggestion';
import TripNavigation from '../Components/TripNavigation';
import HistoryNav from '../Components/HistoryNav';
import CommentScreen from '../Components/CommentScreen';
import MenuChat from '../Components/reactChatScreen/MenuScreen';
import ScreenChat from '../Components/reactChatScreen/ChatScreen';
import NewConversation from '../Components/reactChatScreen/NewConversation';
import Notation from '../Components/Notation';
import Menu from '../Components/Menu';
import Socket from '../Components/Socket';
import CourseEvaluation from '../Components/CourseEvaluation';
import CourseSettings from '../Components/CourseSettings';
import PartenaireScreen from '../Components/PartenairePage';
import SettingPartenaire from '../Components/SettingPartenaire';

const Stack = createStackNavigator();



function Home() {
    return (
        <Stack.Navigator
            screenOptions={{
            headerShown: false,
            }}

        >
            <Stack.Screen
                  name="HomePage"
                  component={HomePage}
            />
            <Stack.Screen
                  name="CommentScreen"
                  component={CommentScreen}
            />
            <Stack.Screen
                  name="TagSelection"
                  component={TagSelection}
            />
            <Stack.Screen
                  name="SettingPartenaire"
                  component={SettingPartenaire}
            />
        </Stack.Navigator>
    );
}

export default Home;
