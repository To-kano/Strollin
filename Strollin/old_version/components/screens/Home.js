import React, { Component } from 'react';
import {
  Text, View, TouchableHighlight, FlatList, Button, ImageBackground, StyleSheet, Dimensions
} from 'react-native';
import stylesHomepage from '../../styles/homepage';
import stylesGeneric from '../../styles/genericStyle';
import { RondFormeText } from '../../features/geoForme/rondForm';

class ElementHistoryNav extends Component {
  constructor(props) {
    super(props);
    this.state = { showMap: false };
  }

  showMap = () => {
    this.setState({
      showMap: !this.state.showMap
    });
  }

  render() {
    if (this.state.showMap == false) {
      return (
        <View style={{
          margin: 20, flex: 1, alignItems: 'center', justifyContent: 'space-evenly'
        }}
        >
          <TouchableHighlight onPress={this.showMap}>
            <View style={{ flexDirection: 'row', flex: 1 }}>
              <Text style={[{ fontWeight: 'bold', fontSize: 20, margin: 15 }]}>
                {this.props.tag}
              </Text>
              <View style={{ backgroundColor: '#39A5D6', borderRadius: 20 }}>
                <Text style={{ fontSize: 20, margin: 15 }}>Bouvelard Saint-Germain</Text>
                <View style={{ flexDirection: 'row-reverse' }}>
                  <Text style={[{
                    textAlign: 'left', fontSize: 15, marginBottom: 20, marginRight: 15
                  }]}
                  >
                    Paris
                  </Text>
                </View>
                <View
                  // eslint-disable-next-line react-native/no-inline-styles
                  style={{
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                  }}
                >
                  <Text style={[{ fontSize: 20, margin: 15 }]}>1,5 H - 2,8 kM</Text>
                  <Text style={[{ fontSize: 20, margin: 15 }]}>3 Pins</Text>
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      );
    }
    return (
      <View>
        <TouchableHighlight onPress={this.showMap}>
          <Text>Map</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

export default class Home extends React.Component {
  data = [{ key: 1 }, { key: 2 }, { key: 3 }, { key: 4 }, { key: 5 }, { key: 6 }];

  render() {
    // <Text style={[{ textAlign: "right", marginRight: 20, marginBottom: 5, fontSize: 20 }]}>Your previous trip</Text>

    //  <FlatList
    //  data={this.data}
    //  renderItem={({ item }) => <ElementHistoryNav tag={item.key} />}
    /// >

    const { navigation } = this.props;
    console.disableYellowBox = true;

    return (
      <View style={{ flex: 1 }}>
        <View style={[{ flex: 1, backgroundColor: 'white' }]}>
          <ImageBackground
            source={require('../../assets/login_image.jpg')}
            style={{
              width: Dimensions.get('window').width, height: Dimensions.get('window').width / 1.8, justifyContent: 'center', flex: 1
            }}
          >
            <View>
              <View style={[{ flexDirection: 'row', justifyContent: 'center' }]}>
                <RondFormeText text="Strollin'" size={140} />
              </View>
              <View style={[{
                backgroundColor: 'white', width: '120%', height: '32%', rotation: -5, display: 'none'
              }]}
              />
            </View>
          </ImageBackground>
        </View>
        <View style={{ flex: 1.5, marginHorizontal: '35%', backgroundColor: 'white' }}>
          <Button
            title="Log Out"
            color="#89B3D9"
            onPress={() => this.props.navigation.navigate('UserLogin')}
          />
        </View>
        <View style={[{ flex: 0.5, backgroundColor: 'white', justifyContent: 'center' }]}>
          <Button
            color="#D99860"
            title="New Trip"
            onPress={() => this.props.navigation.navigate('TripSuggestion', { uid: navigation.getParam('uid') })}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    // justifyContent: 'center',
    // backgroundColor: "gray"
  },
  center: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    // backgroundColor: "gray"
  }
});
