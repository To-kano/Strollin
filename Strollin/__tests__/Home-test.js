import React from 'react';
import {shallow} from 'enzyme';
import Home from '../Components/Home';
import renderer from 'react-test-renderer';

import {
    View, Button
  } from 'react-native';

const navigationTest = {
    navigate: (test) => jest.fn()
}


describe('Home', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<Home navigation={navigationTest}/>)
            expect(component).toMatchSnapshot()
        });

        test('test render Home', () => {
            renderer.create(
                <Home navigation={navigationTest}/>,
            )      
          })

          it('handles button', () => {

            const wrapper =  shallow(
                    <Home navigation={navigationTest} />
            );

            expect(wrapper.find(Button).length).toBe(3);

            wrapper
            .find(Button)
            .at(0)
            .props()
            .onPress()

            wrapper
            .find(Button)
            .at(1)
            .props()
            .onPress()

            wrapper
            .find(Button)
            .at(2)
            .props()
            .onPress()
      
          });
          
    });
});