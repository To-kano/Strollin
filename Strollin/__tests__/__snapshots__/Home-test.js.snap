// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
